### ISSUE TITLE
setPin() causes the element to center

### ISSUE THREAD

**Issue by vitalybe** [janpaepke/ScrollMagic#378]
Consider the following jsbin:

http://jsbin.com/nunugumemi/edit?html,js,output

As soon as "setPin()" is called, the target element gets "margin: auto" which cause it to jump to the center. When the scroll is triggered, it jumps back to left, and then after the duration, jumps again back to center.

Is that by design?

**Comment by blueForestIcarus** [comment]
I have had problems with this as well. I just worked around it by using "!important" but its not a very clean solution. 

Is there a reason for this, if so it should be done in a different way which does not alter the appearance of the page. At the very least there should be an option in setPin().

**Comment by krnlde** [comment]
Create a full width container surrounding your `#animate1` and apply the `.setPin()` to that one.

Explanation: In `position: fixed` (which is applied when pinning) there won't be a valid `margin: auto`

**Comment by blueForestIcarus** [comment]
>Create a full width container surrounding your #animate1 and apply the .setPin() to that one.

This does not address the original issue, which was: why does calling "setPin" give the target  element the style "margin: auto". 

>Explanation: In position: fixed (which is applied when pinning) there won't be a valid margin: auto

Yes, exactly. As we have establish that said style is inherently ineffective while an element is pinned, the fact that the library sets "margin: auto" when calling set pin should be considered a bug.

**Comment by aaronstezycki** [comment]
Yeah, for me styling should be set at the .css level and not assume margin is needed. Would be nice to get that removed, as well as the 'inset' property.

**Comment by janpaepke** [comment]
Sorry for the decade-long wait.

Yes, this was a known v2 pinning side effect. `setPin()` wrapped the target in a spacer div and applied `margin: auto` to the pinned element's inline styles for centering consistency. If your element wasn't centered to begin with, this caused an unwanted jump. The workaround was to override the inline margin after pinning, but it was fragile.

v3 has no pinning system and never modifies element styles. CSS `position: sticky` is the recommended replacement.

---
**ScrollMagic v3 is now available** -- a complete rewrite with an entirely new architecture and API. v2 will not receive further updates (2.0.9 is the final release).

👉 `npm install scrollmagic@next` -- [Docs](https://github.com/janpaepke/ScrollMagic/blob/main/README.md) · [Release notes](https://github.com/janpaepke/ScrollMagic/releases/tag/v3.0.0-beta.0) · [npm](https://www.npmjs.com/package/scrollmagic/v/3.0.0-beta.0)
